jl4/app/Main.hs

```diff
@@ -41,13 +47,28 @@ main = do
       _ <- Shake.use Rules.EvaluateLazy uri
       mep <- Shake.use Rules.ExactPrint uri
       case (mtc, mep) of
-        (Just tcRes, Just ep) | tcRes.success -> logWith recorder Info $ ExactPrint ep
-        (_, _) -> logWith recorder Error $ CheckFailed uri
+        (Just tcRes, Just ep)
+          | tcRes.success -> logWith recorder Info  $ if options.verbose
+                                                      then ExactPrint ep
+                                                      else SuccessOnly
+        (_, _)            -> do
+          logWith    recorder Error $ CheckFailed uri
+          logWith errRecorder Error $ CheckFailed uri
 
-newtype Options = MkOptions {files :: NonEmpty FilePath}
+  errs <- getErrs
+  if null errs
+  then exitSuccess
+  else exitFailure
+
+data Options = MkOptions
+  { files :: NonEmpty FilePath
+  , verbose :: Bool }
 
 optionsDescription :: Options.Parser Options
-optionsDescription = MkOptions <$> some1 (strArgument (metavar "FILES..."))
+optionsDescription = MkOptions
+  <$> some1 (strArgument (metavar "FILES..."))
+  <*> switch (long "verbose" <> short 'v' <> help "Enable verbose output")
```

Comment:
I think if you add a `help` option, then that option should say what it does, exactly, i.e. it also changes the way the exit codes behave.